### PR TITLE
ENG-652 go-service/uuid package- IsValid() fails with an all 0 uuid 00000000-0000-0000-0000-000000000000

### PR DESCRIPTION
**Pull request by gudetimes1234**

### CODE REVIEW

**Comment by gudetimes1234** [comment]
Looks like tests are failing because Circle isn't able to spin up the environment

File: uuid/validate.go [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 // Note that this function was copied from https://stackoverflow.com/questions/25051675/how-to-validate-uuid-v4-in-go
 func IsValid(uuid string) bool {
-	r := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
-	return r.MatchString(uuid)
+	var isValid bool
+	_, err := googleuuid.Parse(uuid)
+	if err == nil {
+		isValid = true
+	}
+	return isValid
```
Comment (tsharkey) [inline uuid/validate.go]: You can make this a little shorter and just do `return err == nil`

**Comment by tsharkey** [comment]
Just noticed that we don't have any tests for this stuff either. Can you add some tests for this function as well please

File: uuid/validate_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package uuid
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsValid(t *testing.T) {
```
Comment (tsharkey) [inline uuid/validate_test.go]: For our test names we try to prefix them with what type of test they are. So in this case `TestUnit_IsValid`. We do this so we can filter tests when running them from our Makefiles. Like just trying to run Unit tests you can do `go test -r Unit` and then all the unit tests run. 
```suggestion
func TestUnit_IsValid(t *testing.T) {
```

**Review by tsharkey — CHANGES_REQUESTED** [review]

**Review by tsharkey — APPROVED** [review]

**Pull request merged**